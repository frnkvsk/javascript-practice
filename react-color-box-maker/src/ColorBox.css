.ColorBox {
  display: flex;
  flex-direction: row;
  align-content: center;
  justify-items: center;
  /* width: 100%; */
  /* height: 20px; */
  border: 1px solid black; 
}
/* Box {
  margin: 0;
} */
/* .btn {  
  width: 20px;
  height: 20px;
  margin: 0;
  padding: 0;
} */
/* .btn:hover {
  bac
} */

.hidden {
  display: none;
}