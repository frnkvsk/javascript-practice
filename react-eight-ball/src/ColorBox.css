
.ColorBox {
  width: 500px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
}

.ColorBox-ul {
  display: flex;
  width: 100%;
  list-style: none;
  padding: 0;
  margin: 5;
}

.ColorBox-ul li {
  width: 32px;
  height: 30px;
  border: .3px solid white;
}

.ColorBox-button {
  width: 100px;
  font-size: 26px;
  margin-bottom: 20px;
  color: white;
  background: rgb(61, 201, 103);
  border-radius: 5px;
  border-style: none;
}

.ColorBox-button:hover {
  cursor: pointer;
  background: rgb(32, 236, 93);
}
.red{background:red}
.green{background:green}
.blue{background:blue}
.orange{background:orange}
.cyan{background:cyan}
.purple{background:purple}
.yellow{background:yellow}
.chartreuse{background:chartreuse}
.aqua{background:aqua}
.blueviolet{background:blueviolet}
.cornflowerblue{background:cornflowerblue}
.greenyellow{background:greenyellow}
.gold{background:gold}
.violet{background:violet}
.skyblue{background:skyblue}
.salmon{background:salmon}