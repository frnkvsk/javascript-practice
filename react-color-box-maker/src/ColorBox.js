import React from 'react';
import './ColorBox.css';
// import NewBoxForm from './NewBoxForm';
import { styled } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';
import { red } from '@material-ui/core/colors';




function ColorBox({id, backgroundColor, width, height, hideBox}) {
  const useStyles = makeStyles((theme) => ({
    container: {
      // display: 'grid',
      // gridTemplateRows: '1fr, 1fr',
      // gridGap: theme.spacing(1),
      display: 'flex',
      justify: 'flex-start',
      alignItems: 'flex-start',
      width: width + 100 + "px",
    },
    box: {
      // margin: theme.spacing(0),
      
      display: 'flex',
      justify: 'flex-start',
      alignItems: 'flex-start',
      textAlign: 'center',
      // backgroundColor: theme.palette.text.secondary,
      width: '0',
      height: '35px',
      color: 'red',
      marginLeft: '5px',
      border: '1px solid red',
    },
  }));
  // const Box1 = styled(Box)({
  //   width: width + 100 + "px",
  //   layout: flex
  // });
  const Box2 = styled(Box)({
    backgroundColor: backgroundColor,
    width: width + "px",
    height: height + "px"
  });
  const removeThisBox = () => {
    hideBox(id)
  }
  const classes = useStyles();
  return (
    <div className="ColorBox" style={{width:(+width+70)+"px"}}>
      <Grid className={classes.container} >
        <Box2 className="ColorBox-Box2"/>
        <Button className={classes.box} onClick={removeThisBox}>X</Button>
      </Grid>
    </div>
    
  );
}

export default ColorBox;