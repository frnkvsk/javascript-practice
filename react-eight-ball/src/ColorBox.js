import React, {useState} from 'react';
import './ColorBox.css';

const ColorBox = () => {
  const colors = ["red", "green", "blue", "orange", "cyan", "purple", "yellow", "chartreuse", "aqua", "blueviolet", "cornflowerblue", "greenyellow", "gold", "violet", "skyblue", "salmon"];
  let [boxes, setBoxes] = useState(colors.map((e,i) => <li key={e+i} className={e}></li>) );
  const random = () => Math.floor(Math.random() * colors.length);
  const colorNextBox = () => {    
    let rand = random();
    boxes[rand] = <li key={boxes[rand].key} className={colors[random()]}></li>;
    setBoxes(boxes.map(e => e));
  }
  return (
    <div className="ColorBox">
      <ul className="ColorBox-ul">
        {boxes}
      </ul>
      <button className="ColorBox-button" onClick={colorNextBox}>Change</button>
    </div>
  )
}

export default ColorBox
