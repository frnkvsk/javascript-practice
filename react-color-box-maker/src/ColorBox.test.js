import React from 'react';
import { render } from '@testing-library/react';
import ColorBox from './ColorBox';

// smoke test
it("Control renders without crashing", () => {
  render(<ColorBox />);
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<ColorBox />);
  expect(asFragment()).toMatchSnapshot();
});
