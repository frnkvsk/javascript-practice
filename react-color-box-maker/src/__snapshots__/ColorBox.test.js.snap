// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorBox"
  >
    <div
      class="MuiGrid-root makeStyles-container-5"
    >
      <div
        class="MuiBox-root MuiBox-root-8 Styled(MuiBox)-root-7 ColorBox-Box2"
      />
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-box-6"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          X
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
